web/src/hooks/use-persistence.ts

```diff
@@ -1,21 +1,21 @@
 import { useEffect, useState, useCallback } from "react";
 import { get as getData, set as setData } from "idb-keyval";
 
-type usePersistenceReturn = [
-  value: any | undefined,
-  setValue: (value: string | boolean) => void,
+type usePersistenceReturn<S extends any> = [
+  value: S | undefined,
+  setValue: (value: S) => void,
   loaded: boolean,
 ];
 
-export function usePersistence(
+export function usePersistence<S>(
   key: string,
-  defaultValue: any | undefined = undefined
-): usePersistenceReturn {
+  defaultValue: S | undefined = undefined
+): usePersistenceReturn<S> {
   const [value, setInternalValue] = useState<any | undefined>(defaultValue);
```

Comment:
Could this be `const [value, setInternalValue] = useState<S | undefined>(defaultValue);`?